temporal-sdk/src/main/java/io/temporal/client/WorkflowClient.java

```diff
@@ -858,299 +858,668 @@ static <R, A1, A2, A3, A4, A5, A6> WorkflowUpdateHandle<R> startUpdate(
+   * @param arg5 fifth update function parameter
+   * @param withStartOperation start workflow operation
+   * @return update result
+   */
+  @Experimental
+  static <R, A1, A2, A3, A4, A5> R executeUpdateWithStart(
+      Proc5<A1, A2, A3, A4, A5> updateMethod,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5,
+      @Nonnull UpdateOptions<R> options,
+      @Nonnull WithStartWorkflowOperation<?> withStartOperation) {
+    return WorkflowClientInternalImpl.executeUpdateWithStart(
+        () -> updateMethod.apply(arg1, arg2, arg3, arg4, arg5), options, withStartOperation);
+  }
+
+  /**
+   * Execute a six argument update workflow request synchronously, along with a workflow start
+   * request.
+   *
+   * @param updateMethod The only supported value is method reference to a proxy created through
+   *     {@link #newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first update function parameter
+   * @param arg2 second update function parameter
+   * @param arg3 third update function parameter
+   * @param arg4 fourth update function parameter
+   * @param arg5 fifth update function parameter
+   * @param arg6 sixth update function parameter
+   * @param withStartOperation start workflow operation
+   * @return update result
+   */
+  @Experimental
+  static <R, A1, A2, A3, A4, A5, A6> R executeUpdateWithStart(
+      Proc6<A1, A2, A3, A4, A5, A6> updateMethod,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5,
+      A6 arg6,
+      @Nonnull UpdateOptions<R> options,
+      @Nonnull WithStartWorkflowOperation<?> withStartOperation) {
+    return WorkflowClientInternalImpl.executeUpdateWithStart(
+        () -> updateMethod.apply(arg1, arg2, arg3, arg4, arg5, arg6), options, withStartOperation);
+  }
+
+  /**
+   * Executes zero argument workflow.
+   *
+   * @param updateMethod The only supported value is method reference to a proxy created through
+   *     {@link #newWorkflowStub(Class, WorkflowOptions)}.
+   * @param withStartOperation start workflow operation
+   * @return update result
+   */
+  @Experimental
+  static <R> R executeUpdateWithStart(
+      Func<R> updateMethod,
+      @Nonnull UpdateOptions<R> options,
+      @Nonnull WithStartWorkflowOperation<?> withStartOperation) {
+    return WorkflowClientInternalImpl.executeUpdateWithStart(
+        updateMethod::apply, options, withStartOperation);
+  }
+
+  /**
+   * Executes one argument workflow together with an update workflow request.
+   *
+   * @param updateMethod The only supported value is method reference to a proxy created through
+   *     {@link #newWorkflowStub(Class, WorkflowOptions)}.
+   * @param arg1 first workflow argument
+   * @param withStartOperation start workflow operation
+   * @return update result
+   */
+  @Experimental
+  static <A1, R> R executeUpdateWithStart(
+      Func1<A1, R> updateMethod,
+      A1 arg1,
+      @Nonnull UpdateOptions<R> options,
```

Comment:
It is a bit odd to use `UpdateOptions` here since almost none of the `options` here make any sense to set... The only option is makes sense to set is the update ID.

Reply:
I guess when we have durable on admitted support we will need to have a `WaitStage` parameter for that...

Reply:
That's absolutely true. I'd support simplifying this `executeUpdateWithStart(updateMethod, args ..., startOp)`. We can always add a new overload when we need it.

Reply:
I really want to avoid adding overloads to these methods if we can avoid it because the number of methods will explode with all the combinations for different numbers of parameters. I would support adding a `TypedUpdateOptions` that only takes `updateID` and `WaitStage` and is used for all the update APIs on `WorkflowClient`